src/shared/conf-parser.c

```diff
@@ -1285,6 +1285,45 @@ int config_parse_permille(
         return 0;
 }
 
+int config_parse_io_cost_qos_scaling_percentage(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint32_t *io_cost_qos_scaling_percentage = data;
+        double result;
```

Comment:
You probably don't want to use a double here. We typically avoid using floating point types and use things like fixed point (i.e. in the case of systemd-oomd) to store decimal values. systemd-oomd uses loadavg_t which you could likely use here as well

Reply:
I tested this briefly and found that loadavg_t expects a decimal point in the input, however there does not need to be a decimal point in the input (tried writing to /sys/fs/cgroup/io.cost.qos). Linux load average does appear to require a decimal point. 

Would you have any other recommendations? The best option I could think of at the time was double.

Reply:
Sure io.cost.qos doesn't need to be a value with a decimal but there's also no downside to writing it, right? 50% or 50.00%, they are the same.
On the user side we can either require the input to be a 2 decimal place value or tweak the helpers/parsing code to accept values without dots.